physcraper/__init__.py

```diff
@@ -1406,38 +1345,49 @@ def get_rank_info_from_web(self, taxon_name):
                 {"taxon id": ncbi_id, "lineage": lineage, "rank": lineage2ranks, "taxon name": tax_name}
         return ncbi_id
 
-    def find_tax_id(self, otu_dict_entry=None, acc=None):
-        """ Find the taxon id in the  otu_dict entry or of a Genbank accession number if no name is given.
-        If not already known it will ask ncbi using the accession number
-
-        :param otu_dict_entry: otu_dict entry
-        :param acc: Genbank accession number
-        :return: ncbi taxon id
-        """
-        # debug("find tax id")
-        inputinfo = False
-        if otu_dict_entry is not None or acc is not None:
-            inputinfo = True
-        assert inputinfo is True
-        ncbi_id = None
-        if otu_dict_entry:
-            ncbi_id = otu_dict_entry["^ncbi:taxon"]
-        if ncbi_id is None:
-            gb_id = None
-            if acc:
-                gb_id = acc
-            elif "^ncbi:accession" in otu_dict_entry:
-                gb_id = otu_dict_entry["^ncbi:accession"]
-            if gb_id in self.acc_ncbi_dict:
-                ncbi_id = self.acc_ncbi_dict[gb_id]
-            else:
-                read_handle = self.entrez_efetch(gb_id)
-                ncbi_id = get_ncbi_tax_id(read_handle)
-                self.acc_ncbi_dict[gb_id] = ncbi_id
-            assert ncbi_id is not None
+ #removed function find_tax_id because it wasn't being used
+   
+    def get_tax_seq_acc(self, acc=None):
+        gb_id = acc
+        if len(gb_id.split(".")) == 1:
+            debug("accession number {} not recognized".format(gb_id))
+        if gb_id in self.acc_tax_seq_dict:
+            taxname = self.acc_tax_seq_dict[gb_id]["taxname"]
+            ncbi_id = self.acc_tax_seq_dict[gb_id]["^ncbi:taxon"]
+            seq = self.acc_tax_seq_dict[gb_id]["seq"]
+        else:
+            read_handle = self.entrez_efetch(gb_id)
+            tax_name = get_ncbi_tax_name(read_handle)
+            ncbi_id = get_ncbi_tax_id(read_handle)
+            seq = read_handle[0][u'GBSeq_sequence']
+            tax_name = tax_name.replace(" ","_") #TODO check that searches are using names without spaces 
+            self.ncbiid_to_spn[ncbi_id] = tax_name
+            self.acc_ncbi_dict[gb_id] = ncbi_id
+            self.acc_tax_seq_dict[gb_id] = {'taxname':tax_name, "^ncbi:taxon":ncbi_id, 'seq':seq} #This is going to be a memory hog...
```

Comment:
I don't think this dict is necessary. We have the ncbiid_to_spn, acc_ncbi_dict, and spn_to_ncbiid dicts to access taxon information from names, ids and accession numbers and a new_seqs dict which links genbank accession number with the sequence.

Reply:
I agree - these can definitely be merged. The only difference here is currently acc_tax_seq_dict gets the full sequence, while the new seqs dictionary just has the blast matching sequence, but that is temporary.